spec/controllers/my/files_controller_spec.rb

```diff
@@ -1,44 +1,39 @@
 require 'spec_helper'
 
 describe My::FilesController, :type => :controller do
- 
+
   before(:each) do
-    @user = FactoryGirl.find_or_create(:archivist)
-    sign_in @user
+    archivist = FactoryGirl.create(:archivist)
+    sign_in archivist
 
-    GenericFile.destroy_all
-    Collection.destroy_all
-    @other_user = FactoryGirl.find_or_create(:user)
-    @my_file = FactoryGirl.create(:generic_file, depositor: @user)
-    @unshared_file = FactoryGirl.create(:generic_file, depositor: @other_user)
-    @edit_shared_with_me = FactoryGirl.create(:generic_file).tap do |r|
-      r.apply_depositor_metadata @other_user
-      r.edit_users += [@user.user_key]
+    other_user = FactoryGirl.create(:user)
+    @my_file = FactoryGirl.create(:generic_file, depositor: archivist)
+    @unshared_file = FactoryGirl.create(:generic_file, depositor: other_user)
+    @edit_shared_with_me = FactoryGirl.build(:generic_file) do |r|
+      r.apply_depositor_metadata other_user
+      r.edit_users += [archivist.user_key]
       r.save!
     end
-    @read_shared_with_me = FactoryGirl.create(:generic_file).tap do |r|
-      r.apply_depositor_metadata @other_user
-      r.read_users += [@user.user_key]
+    @read_shared_with_me = FactoryGirl.build(:generic_file) do |r|
+      r.apply_depositor_metadata other_user
+      r.read_users += [archivist.user_key]
       r.save!
     end
   end
 
   describe "Logged in user" do
-      it "should respond with success" do
+      it "should respond with success and shows the correct documents" do
         get :index
         expect(response).to be_successful
-      end
-
-      it "shows the correct documents" do
-        get :index
+        document_list_ids = assigns[:document_list].map(&:id)
         # shows documents shared with me
-        expect(assigns[:document_list].map(&:id)).to include(@read_shared_with_me.id)
```

Comment:
Looping over this object everytime to create an array of ID's could be slow? Lets do it once and assign it to a variable, and then use this variable in the asserts?